### PR TITLE
SoundSourceFFmpeg + ReadAheadFrameBuffer: Debugging options

### PR DESCRIPTION
**Pull request by uklotzde**
Minor changes and extensions that will help to analyze https://bugs.launchpad.net/mixxx/+bug/1934785.

### CODE REVIEW

File: src/sources/readaheadframebuffer.cpp [hunk 1]
```diff
@@ -3,10 +3,17 @@
...
 
+// Override or set to `true` to enable verbose debug logging.
 #if !defined(VERBOSE_DEBUG_LOG)
 #define VERBOSE_DEBUG_LOG false
 #endif
 
+// Override or set to `true` to break with a debug assertion
+// if an overlap or gap in the audio stream has been detected.
+#if !defined(DEBUG_ASSERT_ON_DISONTINUITIES)
+#define DEBUG_ASSERT_ON_DISONTINUITIES false
```
Comment (Be-ing) [inline src/sources/readaheadframebuffer.cpp:14]:
```suggestion
#define DEBUG_ASSERT_ON_DISCONTINUITIES false
```

File: src/sources/soundsourceffmpeg.cpp [hunk 2]
```diff
@@ -60,7 +60,7 @@ constexpr int64_t kavStreamDecoderFrameDelayAAC = 2112;
 // Use 0-based sample frame indexing
 constexpr SINT kMinFrameIndex = 0;
 
-constexpr SINT kSamplesPerMP3Frame = 1152;
+constexpr SINT kMaxSamplesPerMP3Frame = 1152;
```
Comment (Be-ing) [inline src/sources/soundsourceffmpeg.cpp:63]: unrelated?

File: src/sources/readaheadframebuffer.cpp (hunk 1 above)
Comment (uklotzde) [reply src/sources/readaheadframebuffer.cpp]: :see_no_evil:

**Comment by Be-ing** [comment]
How do you intend to set this new preprocessor definition? Don't you need to add an option to CMakeLists.txt?

**Comment by uklotzde** [comment]
> How do you intend to set this new preprocessor definition? Don't you need to add an option to CMakeLists.txt?

This would trigger a complete recompile. Setting it manually in the .cpp file is easier for quick testing. Let's leave it as is, not need to pollute CMakeLists.txt. The comment just mentions both options.

**Review by Be-ing — APPROVED** [review]
